### PR TITLE
encryption: e2e w/ real controllers against real cluster with fake deployer

### PR DESCRIPTION
**Pull request by sttts**
This allows us to run exhaustive e2e tests without the 20min slow-down of the static-pod deployer per key rotation.

### CODE REVIEW

File: test/e2e/encryption_quick_test.go [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+	return s, nil
+}
+
+func (c *secretInterceptor) Update(s *corev1.Secret) (*corev1.Secret, error) {
+	s, err := c.SecretInterface.Update(s)
+	if err != nil {
+		return s, err
+	}
+
+	fmt.Println("Update", s.Name)
```
Comment (p0lyn0mial) [inline test/e2e/encryption_quick_test.go]: `t.Log`

File: test/e2e/encryption_quick_test.go [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+	// create APIServer clients
+	fakeConfigClient := configv1clientfake.NewSimpleClientset(&configv1.APIServer{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}})
+	fakeConfigInformer := configv1informers.NewSharedInformerFactory(fakeConfigClient, 10*time.Minute)
+	fakeApiServerClient := fakeConfigClient.ConfigV1().APIServers()
+
+	// create controllers
+	eventRecorder := events.NewLoggingEventRecorder(component)
+	deployer := NewInstantDeployer(stopCh, kubeInformers, kubeClient.CoreV1(), fmt.Sprintf("encryption-config-%s", component))
+
+	controllers, err := encryption.NewControllers(
```
[the remark this answers is not in the document]
Comment (sttts) [reply test/e2e/encryption_quick_test.go]: the component is different (=> different secrets), and I use another operator config and fake APIServer config, for that reason to separate it.

File: test/e2e/encryption_quick_test.go [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+			ps = append(ps, s)
+		}
+		rs = append(rs, fmt.Sprintf("%s=%s", strings.Join(r.Resources, ","), strings.Join(ps, ",")))
+	}
+	return strings.Join(rs, ";")
+}
+
+func NewInstantDeployer(stopCh <-chan struct{}, kubeInformers v1helpers.KubeInformersForNamespaces, secretsClient corev1client.SecretsGetter,
+	secretName string) *lockStepDeployer {
+	return &lockStepDeployer{
```
[the remark this answers is not in the document]
Comment (sttts) [reply test/e2e/encryption_quick_test.go]: done

**Pull request merged**